tests/flytekit/unit/core/test_offloaded_literals.py

```diff
@@ -0,0 +1,179 @@
+from dataclasses import dataclass
+import typing
+
+from mashumaro.mixins.json import DataClassJSONMixin
+import pytest
+from flytekit import task
+from flytekit.configuration import Image, ImageConfig, SerializationSettings
+from flytekit.models import literals as literal_models
+from flytekit.core import context_manager
+from flytekit.models.types import SimpleType
+from flytekit.core.type_engine import TypeEngine
+
+@pytest.fixture
+def flyte_ctx():
+    return context_manager.FlyteContext.current_context()
+
+
+def test_task_offloaded_literal_single_input(tmp_path):
+    @task
+    def t1(a: int) -> str:
+        return str(a)
+
+    original_input_literal = literal_models.Literal(
+        scalar=literal_models.Scalar(primitive=literal_models.Primitive(integer=3))
+    )
+
+    # Write offloaded_lv as bytes to a temp file
+    with open(f"{tmp_path}/offloaded_proto.pb", "wb") as f:
```

Comment:
I'm surprised using `/` works with windows. I usually go with:

```suggestion
    with (tmp_path / "offloaded_proto.pb").open("wb") as f:
```

Reply:
that's a good call. I'm going to fix this in a separate PR. 

As for what happens on windows, my guess is that the `/` becomes part of the file name.